nautobot_golden_config/models.py

```diff
@@ -513,29 +541,9 @@ def clean(self):
             if not str(self.sot_agg_query.query).startswith(GRAPHQL_STR_START):
                 raise ValidationError(f"The GraphQL query must start with exactly `{GRAPHQL_STR_START}`")
 
-        if self.scope:
-            filterset_class = get_filterset_for_model(Device)
-            filterset = filterset_class(self.scope, Device.objects.all())
-
-            if filterset.errors:
-                for key in filterset.errors:
-                    error_message = ", ".join(filterset.errors[key])
-                    raise ValidationError({"scope": f"{key}: {error_message}"})
-
-            filterset_params = set(filterset.get_filters().keys())
-            for key in self.scope.keys():
-                if key not in filterset_params:
-                    raise ValidationError({"scope": f"'{key}' is not a valid filter parameter for Device object"})
-
     def get_queryset(self):
         """Generate a Device QuerySet from the filter."""
-        if not self.scope:
-            return Device.objects.all()
-
-        filterset_class = get_filterset_for_model(Device)
-        filterset = filterset_class(self.scope, Device.objects.all())
-
-        return filterset.qs
+        return self.dynamic_group.get_queryset()
```

Comment:
Just below this there is a `get_filter_as_string` which I think should just point to `get_absolute_url`, as that should be the core way of doing it.

Reply:
Done